### PR TITLE
Multiple votes accumulated on one person

### PR DESCRIPTION
**Pull request by MJJojo97**
See [https://github.com/OpenSlides/OpenSlides/pull/6376](url)

### CODE REVIEW

File: global/meta/models.yml [hunk 1]
```diff
@@ -2683,6 +2683,10 @@ poll:
     type: number
     default: 1
     restriction_mode: A
+  max_votes_per_option:
+    type: number
+    default: 1
```
Comment (jsangmeister) [inline global/meta/models.yml:2688]: Add `minimum: 1` and `required: true`

**Review by jsangmeister — CHANGES_REQUESTED** [review]
The changes itself look good, but some tests for it are missing. You can add them by doing the following (I don't know how familiar you are with pytest and the development setup itself, so I'm trying to be rather more precise):
1) Replace the source in https://github.com/OpenSlides/openslides-backend/blob/a5ea1164f60ce939b4efb0fb51905ce2f5eccbab/dev/docker-compose.dev.yml#L93 with your own fork and branch so you can use the changed vote service code
2) Start the backend dev setup with `make run-dev`
3) Write your tests in the appropriate files (`test_{create|update|vote}.py`). The other tests should provide enough orientation on how to do this; if not, always feel free to ask for help. You should test at least the following things (you can also alter existing tests and just add the field, if appropriate):
  - you can set `max_votes_per_option` in `poll.create` and `poll.update`
  - you cannot set values <0 for the field
  - you cannot update this field if the poll is not in the `Created` state
  - if someone votes multiple times for the same option, the result is correct (note that the tests in `poll_vote.py` automatically start and stop the poll) - without and with vote weights
4) Execute your tests with `pytest` - see their docs for more details, but `pytest -k <test-name>` is most of the time the command of choice to not execute all tests, which is the default
5) Clean up your code by just executing `make` (inside the container!) and fix all errors which flake/mypy report
Also, you'll have to rebase again since the example data changed so the CI can ru successfully. Thank you for you contribution so far and feel free to ask if you run into problems!

File: openslides_backend/action/actions/poll/mixins.py [hunk 2]
```diff
@@ -95,18 +95,25 @@ def on_stop(self, instance: Dict[str, Any]) -> None:
...
                         vote_value = poll["pollmethod"]
-                        vote_weight *= value
-
-                    option_results[option_id][vote_value] += vote_weight
+                        # print("Summing up")
+                        # print(vote_value)
+                        # print(value)
+                        # print(vote_weight)
+                        # print(ballot["weight"])
+                        # vote_weight *= value
```
[the remark this answers is not in the document]
Comment (MJJojo97) [reply openslides_backend/action/actions/poll/mixins.py]: Changed in new commit.

File: global/meta/models.yml (hunk 1 above)
Comment (MJJojo97) [reply global/meta/models.yml:2688]: Changed in new commit.

**Comment by MJJojo97** [comment]
Thank you for the explanations. They were very helpful!

I made all the changes. However, I currently just added the test cases for "<0" as comments because currently you actually can set negative values. Since this holds for values like max_votes_amount, too, I was not sure how I should fix this problem. Could you give me a hint on how I can fix that?

**Review by jsangmeister — APPROVED** [review]
waiting for client. vote service remote has to be changed before merging

File: dev/docker-compose.dev.yml [hunk 3]
```diff
@@ -90,7 +90,7 @@ services:
             - cache
     vote:
         build:
-            context: "https://github.com/OpenSlides/openslides-vote-service.git#main"
+            context: "https://github.com/MJJojo97/openslides-vote-service.git#main"
```
Comment (ostcar) [inline dev/docker-compose.dev.yml:93]: Is it possible to undo this line and still merge this PR?

File: tests/system/action/poll/test_create.py [hunk 4]
```diff
@@ -828,3 +832,26 @@ def test_create_permissions_topic(self) -> None:
...
+    #             "global_abstain": True,
+    #             "onehundred_percent_base": "Y",
+    #             "min_votes_amount": 5,
+    #             "max_votes_amount": 10,
+    #             "max_votes_per_option": -1,
+    #             "backend": "long",
+    #         },
+    #     )
+    #     self.assert_status_code(response, 400)
+    #     self.assert_model_not_exists("poll/1")
```
Comment (ostcar) [inline tests/system/action/poll/test_create.py:857]: Can you either remove this test or uncomment it?

**Review by ostcar — CHANGES_REQUESTED** [review]

**Comment by ostcar** [comment]
Sry. Wrong click :(

**Comment by MJJojo97** [comment]
I opened a new PR for the models.yml changes.

File: dev/docker-compose.dev.yml (hunk 3 above)
Comment (MJJojo97) [reply dev/docker-compose.dev.yml]: I removed this change. Can you try whether it is possible to merge it, now? Or what shall I do for testing that?

**Comment by ostcar** [comment]
> waiting for client. vote service remote has to be changed before merging
The vote service is updated. We don't have to wait for the client, since the old client should be compatible with this change.

**Comment by emanuelschuetze** [comment]
@ostcar Client is ready for merge. Can we merge this backend PR?

**Comment by ostcar** [comment]
I don't want to give a review in the backend since I don't have enough knowledge.

If the backend developers approve it, then it can be merged.

**Review by emanuelschuetze — APPROVED** [review]
Tested with current client PR #904

**Pull request merged**